include/raft.h

```diff
@@ -223,20 +249,58 @@ typedef int (
     msg_appendentries_t* msg
     );
 
-/** 
- * Log compaction
- * Callback for telling the user to send a snapshot.
- *
- * @param[in] raft Raft server making this callback
+/** Callback for sending InstallSnapshot request messages.
+ * @param[in] raft The Raft server making this callback
  * @param[in] user_data User data that is passed from Raft server
- * @param[in] node Node's ID that needs a snapshot sent to
- **/
+ * @param[in] node The node's ID that we are sending this message to
+ * @param[in] msg The InstallSnapshot message to be sent
+ * @return 0 on success */
 typedef int (
-*func_send_snapshot_f
+*func_send_installsnapshot_f
 )   (
     raft_server_t* raft,
     void *user_data,
-    raft_node_t* node
+    raft_node_t* node,
+    msg_installsnapshot_t* msg
+    );
+
+/** Callback for receiving InstallSnapshot request messages.
+ * @param[in] raft The Raft server making this callback
+ * @param[in] user_data User data that is passed from Raft server
+ * @param[in] node The node's ID that we are sending this message to
```

Comment:
This should be the node we are receiving from.

Reply:
The node object, of course. I'll fixed this for the new callbacks as well as the existing ones.